scripts/workspace-conversion/workspace-to-dataset.py

```diff
@@ -0,0 +1,248 @@
+    "Medical imaging _MRI": "TerraCoreValueSets:MedicalImaging_MRI",
+    "Medical imaging_PET": "TerraCoreValueSets:MedicalImaging_PET",
+    "Medical imaging _Xray": "TerraCoreValueSets:MedicalImaging_Xray",
+    "Metabolomic": "TerraCoreValueSets:metabolomic",
+    "Microbiome": "TerraCoreValueSets:Microbiome",
+    "Metagenomic": "TerraCoreValueSets:Metagenomic",
+    "Proteomic": "TerraCoreValueSets:Proteomic",
+    "Transcriptomic": "TerraCoreValueSets:Transcriptomic",
+    "SpatialTranscriptomics": "TerraCoreValueSets:SpatialTranscriptomics",
+    "Trascriptomic_Targeted": "TerraCoreValueSets:Transcriptomic_Targeted",
+    "Trascriptomic_NonTargeted": "TerraCoreValueSets:Transcriptomic_NonTargeted",
+    "Trascriptomic_NonTargeted_RnaSeq": "TerraCoreValueSets:Transcriptomic_NoneTargeted_RnaSeq",
+    "Trascriptomic_NonTargeted_MicroRnaCounts": "TerraCoreValueSets:Transcriptomic_NonTargeted_MicroRnaCounts",
+    "Electrocardiogram": "TerraCoreValueSets:Electrocardiogram",
+}
+
+
+def log_response(response, message):
+    if response.ok:
+        print("success!", response.text)
+    else:
+        print("---------------------------------------------------")
+        print(f"There was a problem running this request: ", message)
+        print(json.loads(response.text)["message"])
+        print("\nRequest:")
+        print(f"\t{response.request.url}")
+        print(f"\t{response.request.headers}")
+        print(f"\t{response.request.body}")
+        print("---------------------------------------------------")
+
+
+def get_access_token():
+    proc = subprocess.Popen(
+        [f"gcloud auth login {user} --brief"], stdout=subprocess.PIPE, shell=True
+    )
+    proc.communicate()
+
+    printProc = subprocess.Popen(
+        ["gcloud auth print-access-token"], stdout=subprocess.PIPE, shell=True
+    )
+    (out, err) = printProc.communicate()
+
+    return out.decode("ASCII").strip()
+
+
+def get_workspace(accessToken):
+    print("Getting workspace information from rawls...")
+    headers = {
+        "Authorization": f"Bearer {accessToken}",
+        "Content-Type": "application/json",
+    }
+
+    # Get dataset list once in if lowerPolicy == of collision later
+    response = requests.get(
+        f"{urlWorkspace}/{workspaceNamespace}/{workspaceName}", headers=headers
+    )
+    responseData = json.loads(response.text)
+    print("loaded workspace information from rawls")
+    return responseData
+
+
+def map_dataset_release_policy(policyString):
+    lowerPolicy = policyString.lower()
+    return policyMap.get(lowerPolicy, policyString)
+
+
+def map_data_modality(modalityArray):
+    ret = []
+    for modality in modalityArray:
+        dataModalityMap.get(modality, None)
+    return list(filter(None, ret))
+
+
+def generate_catalog_metadata(workspace):
+    print("Generating workspace metadata")
+
+    wsAttributes = workspace["workspace"]["attributes"]
+
+    # Set empty up the major objects
+    metadata = {}
```

Comment:
Would there be any way to structure the code for this object in a clearer way so it's obvious at a glance what the nesting is for various fields? For example: https://github.com/DataBiosphere/terra-data-catalog/blob/main/scripts/hca/create-hca-collection.py#L175

Reply:
done!